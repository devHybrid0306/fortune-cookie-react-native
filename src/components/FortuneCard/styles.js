import { StyleSheet } from 'react-native';
import { RatioW } from '../../utils';

const styles = StyleSheet.create({
  container: {
    height: 150 * RatioW,
    padding: 20 * RatioW,
    backgroundColor: '#0ab5ff',
    borderRadius: 20 * RatioW,
    marginBottom: 10 * RatioW,
    marginLeft: 3 * RatioW,
    marginRight: 3 * RatioW,
    alignSelf: 'center',
    justifyContent: 'space-between',
  },
  dateContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(255, 255, 255, .2)',
    width: 120 * RatioW,
    borderRadius: 10 * RatioW,
  },
  title: {
    fontSize: 15 * RatioW,
    color: '#ffffff',
    marginBottom: 20 * RatioW,
  },
  date: {
    fontSize: 12 * RatioW,
    color: '#ffffff',
    textAlign: 'center',
    padding: 3 * RatioW,
  },
});

export default styles;
