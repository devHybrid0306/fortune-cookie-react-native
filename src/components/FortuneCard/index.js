import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

import { getFormattedDate, getRandomColor, RatioW, width } from '../../utils';
import styles from './styles';

const FortuneCard = ({ text, date, isFull, handleCard }) => {
  const mWidth = isFull ? width - 40 * RatioW : (width - 50 * RatioW) / 2;
  const mHeight = isFull ? 150 * RatioW : 220 * RatioW;

  const mColor = getRandomColor();

  return (
    <TouchableOpacity onPress={() => handleCard({ text, date, mColor })}>
      <View
        style={{
          ...styles.container,
          width: mWidth,
          height: mHeight,
          backgroundColor: mColor,
        }}>
        <Text style={styles.title}>{text}</Text>
        <View style={styles.dateContainer}>
          <Text style={styles.date}>{getFormattedDate(date)}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default FortuneCard;
