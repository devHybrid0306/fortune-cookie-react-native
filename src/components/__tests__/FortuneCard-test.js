import React from 'react';
import renderer from 'react-test-renderer';
import FortuneCard from './../FortuneCard';

test('renders correctly', () => {
  const tree = renderer.create(<FortuneCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
