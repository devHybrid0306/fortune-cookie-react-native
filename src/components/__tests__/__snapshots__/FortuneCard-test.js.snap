// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgb(157,79,213)",
        "borderRadius": 40,
        "height": 440,
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginLeft": 6,
        "marginRight": 6,
        "padding": 40,
        "width": 325,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 30,
          "marginBottom": 40,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, .2)",
          "borderRadius": 20,
          "justifyContent": "center",
          "width": 240,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 24,
            "padding": 6,
            "textAlign": "center",
          }
        }
      >
        July 7 2021
      </Text>
    </View>
  </View>
</View>
`;
